lib/ts/controllers/s-table.ts

```diff
@@ -142,7 +142,7 @@ function getCellSlot(cell: HTMLTableCellElement): number {
     if (!(cell.parentElement && cell.parentElement.parentElement instanceof HTMLTableSectionElement)) {
         throw "invalid table"
     }
-    const result = buildIndexOrGetCellSlot(cell.parentElement.parentElement, cell);
+    const result = buildIndexOrGetCellSlot(cell.parentElement.parentElement as any as HTMLTableSectionElement, cell);
```

Comment:
`any` is a bit of an early escape hatch here.  You can use a [type predicate/guard](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates) to safely convince TS you are looking at the right type.  

We're supposedly already doing that above with this check:

```ts
if (!(cell.parentElement && cell.parentElement.parentElement instanceof HTMLTableSectionElement)) {
    throw "invalid table"
}
```

But perhaps TS remains unconvinced?

We can add this type guard:

```cs
function isHTMLTableSectionElement(el: HTMLElement | null | undefined): el is HTMLTableSectionElement {
    return el instanceof HTMLTableSectionElement;
}
```

You then rewrite like this:

```ts
function getCellSlot(cell: HTMLTableCellElement): number {
    const grandParent = cell.parentElement?.parentElement
    if (!isHTMLTableSectionElement(grandParent)) {
        throw "invalid table"
    }
    const result = buildIndexOrGetCellSlot(grandParent, cell);
    if (typeof result !== "number") {
        throw "shouldn't happen"
    }
    return result
}
```

Reply:
The double cast to any then to the element is definitely an ugly band-aid. 

I tried the type guard approach but I'm still getting some errors from the typescript compiler:
![image](https://user-images.githubusercontent.com/4502154/178804433-faa31357-061c-4ee9-ac28-cd7409973153.png)

The assume type approach also didn't work for me. Not sure if I'm doing something incorrectly:
![image](https://user-images.githubusercontent.com/4502154/178804901-c2f864ac-59f3-4ba0-ac5d-f4f1f124d190.png)

Reply:
Huh, I can't repro your error @mukunku, either in my editor or in [TS Playground Demo](https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABAcwKZQMKoDbYMrZxQAUEO2AXIgBIAqAsgDK0CGARtqlrgKKcC2qMFACUVMCH5tUAJ0QBvALAAoRGsQQEAZygoZLMABMACixlDdAXg3kAdAAczFvqkHCA-A6fCXbqCvVEGGBEYgBCGC06JlYOVDxUaHgwXwtiZH0jU3NhEREFAMD1KAALGTgAd0QAIhgwADcWbBhDRCh2TmrCtQBfbo1tXXMtEGwrRDYQGGxDAEkjVAAPAHkZAHF0bnxCEgyDE28oABobXBEAbn7g0KgAT3tUOBDh0d0wy2tqiSlZavylVRFNSlcpVapaEpwUaGMAAcl0JRY9geYC6gN6-XMUBAMiQLzGKj6ylAkFgCCCUQYzA68USZJSAjSOCo0UYqWEiAAPogJLguYhwIZUMA6qhDGJEDgKTQqbFOAkkgh2boAYEsTikFK6joDGQnjKYjSFfTlZdlESVCTFUhJtM5gsVutNuQCERiFo6ckWbKjZ6lYzhCcRUYtu5vYa4ltlf9+powFo4JxbIRkMR5B7rUdg4Ytj0LoSgA)

If I control click on `HTMLTableSectionElement ` it provides the following source code:

```ts
/** Provides special properties and methods (beyond the HTMLElement interface 
  it also has available to it by inheritance) for manipulating the layout and 
  presentation of sections, that is headers, footers and bodies, in an HTML table. 
*/
interface HTMLTableSectionElement extends HTMLElement {}
```

So it does seem fair that it should be assignable in a predicate.

But I believe the squiggles in your screenshot; I just wonder where they're coming from.